### PR TITLE
mlterm: darwin fixes

### PR DESCRIPTION
**Pull request by Atemu**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/mlterm/default.nix [hunk 1]
```diff
@@ -28,20 +28,20 @@ stdenv.mkDerivation rec {
...
     # Not supported on Darwin
-    vte
     m17n_lib
 
     fcitx5
     fcitx5-gtk
     ibus
-  ] ++ lib.optionals (stdenv.system != "aarch64-linux") [
-    # FIXME Currently broken on aarch64-linux
+  ] ++ lib.optionals (!stdenv.isAarch64 && !stdenv.isDarwin) [
```
Comment (wegank) [inline pkgs/applications/terminal-emulators/mlterm/default.nix:43]: We could just update `uim.meta.broken` and then

```suggestion
  ] ++ lib.optionals (!uim.meta.broken) [
```

Comment (wegank) [reply pkgs/applications/terminal-emulators/mlterm/default.nix:43]: Also, `uim` actually compiles on `aarch64-linux`...

**Review by wegank — APPROVED** [review]
Just a squash away.

**Comment by Atemu** [comment]
@ramkromberg do you still maintain mlterm?

**Comment by Atemu** [comment]
Thanks a bunch @wegank!

**Comment by 7c6f434c** [comment]
… and now `mlterm` with `libssh2` breaks without `openssl` again. I want to readd it and write something about its removal not being a good idea, how to word it? (Or should we just disable libssh2 by default?)

**Comment by Atemu** [comment]
I'm a bit confused what you're trying to say. Could you reword that?

Does libssh2 get fixed by the PR making openssl a propagated dep of libssh2?

**Comment by 7c6f434c** [comment]
https://hydra.nixos.org/build/223620190

I have looked at the code; as mlterm uses openssl directly, I think it is reasonable to list openssl as a dependency. (And mlterm is listed as still-failing in the libssh2 PR)

**Pull request merged**